daffodil-core/src/main/scala/org/apache/daffodil/dsom/RawCommonRuntimeValuedPropertiesMixin.scala

```diff
@@ -49,69 +49,68 @@ trait RawElementRuntimeValuedPropertiesMixin
   protected final lazy val separatorRaw = requireProperty(optionSeparatorRaw)
 }
 
 trait RawLayeringRuntimeValuedPropertiesMixin
   extends PropertyMixin {
-  protected final lazy val optionLayerTransformRaw = findPropertyOption("layerTransform")
+  protected final lazy val optionLayerTransformRaw = findPropertyOption("layerTransform", expressionAllowed = true)
   protected final lazy val layerTransformRaw = requireProperty(optionLayerTransformRaw)
-  protected final lazy val optionLayerEncodingRaw = findPropertyOption("layerEncoding")
+  protected final lazy val optionLayerEncodingRaw = findPropertyOption("layerEncoding", expressionAllowed = true)
   protected final lazy val layerEncodingRaw = requireProperty(optionLayerEncodingRaw)
-  protected final lazy val optionLayerLengthRaw = findPropertyOption("layerLength")
+  protected final lazy val optionLayerLengthRaw = findPropertyOption("layerLength", expressionAllowed = true)
   protected final lazy val layerLengthRaw = requireProperty(optionLayerLengthRaw)
-  protected final lazy val optionLayerBoundaryMarkRaw = findPropertyOption("layerBoundaryMark")
+  protected final lazy val optionLayerBoundaryMarkRaw = findPropertyOption("layerBoundaryMark", expressionAllowed = true)
   protected final lazy val layerBoundaryMarkRaw = requireProperty(optionLayerBoundaryMarkRaw)
 
 }
 
 trait RawEscapeSchemeRuntimeValuedPropertiesMixin
   extends PropertyMixin {
-
   // package private because used in unit test
-  final lazy val optionEscapeCharacterRaw = findPropertyOption("escapeCharacter")
+  final lazy val optionEscapeCharacterRaw = findPropertyOption("escapeCharacter", expressionAllowed = true)
   private[dsom] final lazy val escapeCharacterRaw = requireProperty(optionEscapeCharacterRaw)
-  final lazy val optionEscapeEscapeCharacterRaw = findPropertyOption("escapeEscapeCharacter")
+  final lazy val optionEscapeEscapeCharacterRaw = findPropertyOption("escapeEscapeCharacter", expressionAllowed = true)
   protected final lazy val escapeEscapeCharacterRaw = requireProperty(optionEscapeEscapeCharacterRaw)
-  protected final lazy val optionEscapeBlockStartRaw = findPropertyOption("escapeBlockStart")
+  protected final lazy val optionEscapeBlockStartRaw = findPropertyOption("escapeBlockStart", expressionAllowed = true)
   protected final lazy val escapeBlockStartRaw = requireProperty(optionEscapeBlockStartRaw)
-  protected final lazy val optionEscapeBlockEndRaw = findPropertyOption("escapeBlockEnd")
+  protected final lazy val optionEscapeBlockEndRaw = findPropertyOption("escapeBlockEnd", expressionAllowed = true)
   protected final lazy val escapeBlockEndRaw = requireProperty(optionEscapeBlockEndRaw)
-  protected final lazy val optionExtraEscapedCharactersRaw = findPropertyOption("extraEscapedCharacters")
+  protected final lazy val optionExtraEscapedCharactersRaw = findPropertyOption("extraEscapedCharacters", expressionAllowed = true)
```

Comment:
escapeBlockStart, escapeBlockEnd, and extraEscapedCharacters cannot be expressions:

https://daffodil.apache.org/docs/dfdl/#_Toc398030751